src/articles/ArticlePreview.js

```diff
@@ -10,17 +18,74 @@ export default function ArticleOverview({
   dontShowImage,
   hasExtension,
 }) {
+  const [isOpen, setIsOpen] = useState(false);
+
   let topics = article.topics.split(" ").filter((each) => each !== "");
   let difficulty = Math.round(article.metrics.difficulty * 100) / 10;
 
+  function handleClose() {
+    setIsOpen(false);
+  }
+
+  function handleOpen() {
+    setIsOpen(true);
+  }
+
   function titleLink(article) {
     let open_in_zeeguu = (
       <Link to={`/read/article?id=${article.id}`}>{article.title}</Link>
     );
     let open_externally = (
-      <a target="_blank" rel="noreferrer" href={article.url}>
-        {article.title}
-      </a>
+      //TODO: Refactor into separate components and add styling to the modal and article titles
```

Comment:
Refactoring is a good idea.
However, maybe first test with users to make sure that they understand the wording and flow